service/resources.ts

```diff
@@ -1,17 +1,53 @@
 import { Empty } from "google-protobuf/google/protobuf/empty_pb";
+import { http } from "service";
 import client from "service/client";
+import { Paginated } from "types/paginatedResponse";
 
 export async function getTermsOfService() {
   const res = await client.resources.getTermsOfService(new Empty());
   return res.toObject();
 }
 
-export async function getRegions() {
-  const regions = await client.resources.getRegions(new Empty());
-  return regions.toObject();
+
+export async function getRegions(): Promise<Region[]> {
+    let req =  await http.get<RegionsResponse>("regions");
+    let res = req.results;
+    // Get all regions, not paginated results
+    while (req.next) {
+        const nextReq = await http.get<RegionsResponse>(req.next);
+        res = res.concat(nextReq.results);
+        req = nextReq;
+    }
+    return res;
+}
+interface RegionsResponse {
+    count: number,
+    next: string,
+    results: Region[]
 }
+export interface Region {
+    id: number,
+    code: string,
+    name: string,
+}
+
+export async function getLanguages(): Promise<Language[]> {
+
+    let req =  await http.get<Paginated<Language>>("languages/");
+    let res = req.results;
+    // Get all Languages, not paginated results
+    while (req.next) {
+        const nextReq = await http.get<Paginated<Language>>(req.next);
+        res = res.concat(nextReq.results);
+        req = nextReq;
+    }
+    return res;
+
```

Comment:
Probably one to feedback to the backend team rather a frontend problem - but doing this means it's completely pointless to have pagination on the endpoint as the whole point of the pagination pattern is to prevent too much data being fetched all in one go, usually because the page doesn't need everything at first to function.

But in this case, since you do need the full data set right from the beginning, the endpoint should just return everything in one request. This is trying to do the same but in a much more wasteful manner over multiple requests.